coalib/core/Graphs.py

```diff
@@ -0,0 +1,62 @@
+from coalib.core.CircularDependencyError import CircularDependencyError
+
+
+def traverse_graph(start_nodes, get_successive_nodes,
+                   run_on_edge=lambda prev, nxt: None):
+    """
+    Traverses all edges of a directed acyclic graph once. Detects cyclic graphs
```

Comment:
it is confusing, cyclic graph in acyclic graph not possible.  
Plus we are expecting more than one connected graphs and No, we can't ensure it is traversing every edge exactly once. so, maybe change it to something like - 
`Traverses the 'graph/s'. Detects cycle by raising..`